clustering.Rmd

```diff
@@ -238,75 +294,93 @@ Note, however, that "large" is relative to *both* the scale of the variables for
 
 ### The clustering algorithm
 
-We begin the K-means \index{K-means!algorithm} algorithm by picking K, and uniformly randomly assigning data to the K clusters. 
+We begin the K-means \index{K-means!algorithm} algorithm by picking K, 
+and randomly assigning an equal number of observations 
+to each of the K clusters.
 Then K-means consists of two major steps that attempt to minimize the
 sum of WSSDs over all the clusters, i.e. the \index{WSSD!total} *total WSSD*:
 
 1. **Center update:** Compute the center of each cluster.
 2. **Label update:** Reassign each data point to the cluster with the nearest center.
 
 These two steps are repeated until the cluster assignments no longer change.
-For example, in the penguin data example, our initialization might look like this:
+For example, in the penguin data example, 
+our random initialization might look like this:
+
 <center>
 ```{r 10-toy-kmeans-init, echo = FALSE, message = FALSE, warning = FALSE, fig.height = 4, fig.width = 4.35, fig.cap = "Random initialization of labels."}
-library(gridExtra)
 set.seed(14)
 penguin_data["label"] <- factor(sample(1:3, nrow(penguin_data), replace = T))
 
-plt_lbl <- ggplot(penguin_data, aes(y = bill_length_scaled, x = flipper_length_scaled, colour = label)) +
+plt_lbl <- ggplot(penguin_data, aes(y = bill_length_standardized, x = flipper_length_standardized, colour = label)) +
   geom_point(size = 2) +
-  xlab("Flipper Length (scaled)") +
-  ylab("Bill Length (scaled)") +
+  xlab("Flipper Length (standardized)") +
+  ylab("Bill Length (standardized)") +
   theme(legend.position = "none") + 
   scale_color_manual(values= cbpalette)
 print(plt_lbl)
 ```
 </center>
 
-And the first four iterations of K-means would look like (each row corresponds to an iteration,
+And we show what the the first four iterations of K-means would look like in  
+Figure \@ref(fig:10-toy-kmeans-iter). 
+There each row corresponds to an iteration,
 where the left column depicts the center update, 
-and the right column depicts the reassignment of data to clusters): 
+and the right column depicts the reassignment of data to clusters.
+
 <center>
 **Center Update**  &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;**Label Update**
-```{r 10-toy-kmeans-iter, echo = FALSE, warning = FALSE, fig.height = 4, fig.width = 8}
+```{r 10-toy-kmeans-iter, echo = FALSE, warning = FALSE, fig.height = 16, fig.width = 8, fig.cap = "First four iterations of K-means clustering on the `penguin_data` example data set. Each row corresponds to an iteration,where the left column depicts the center update, and the right column depicts the reassignment of data to clusters. Cluster centers are indicated by larger points that are outlined in black."}
```

Comment:
missing space after comma in the caption

Reply:
fixed